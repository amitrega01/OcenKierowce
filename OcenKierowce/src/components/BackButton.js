import React, { Component } from 'react';
import { Text, View, TouchableOpacity } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
export default class BackButton extends Component {
  render() {
    return (
      <TouchableOpacity
        activeOpacity={0.6}
        onPress={this.props.onPress}
        style={{
          position: 'absolute',
          top: 32,
          left: 16,
        }}>
        <View style={{ flex: 1, flexDirection: 'row', alignItems: 'center' }}>
          <Ionicons name='ios-arrow-back' size={32} color='rgba(0,0,0,0.75)' />
          <Text style={{ color: 'rgba(0,0,0,0.75)', marginLeft: 8 }}>Wr√≥c</Text>
        </View>
      </TouchableOpacity>
    );
  }
}
