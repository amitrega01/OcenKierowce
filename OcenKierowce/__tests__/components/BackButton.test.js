import React from 'react';
import renderer from 'react-test-renderer';

import BackButton from '../../src/components/BackButton';

describe('<BackButton />', () => {
  it('has 1 child', () => {
    const tree = renderer.create(<BackButton />).toJSON();
    expect(tree.children.length).toBe(1);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<BackButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
