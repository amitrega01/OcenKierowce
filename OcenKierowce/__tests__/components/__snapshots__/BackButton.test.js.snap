// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackButton /> renders correctly 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "left": 16,
      "opacity": 1,
      "position": "absolute",
      "top": 32,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
      }
    }
  >
    <Text />
    <Text
      style={
        Object {
          "color": "rgba(0,0,0,0.75)",
          "marginLeft": 8,
        }
      }
    >
      Wr√≥c
    </Text>
  </View>
</View>
`;
